src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -120,9 +120,16 @@ public void CheckTablePrefix(string databaseName, string schemaName, string expe
             queryManagerFactory.Setup(x => x.GetQueryBuilder(It.IsAny<DatabaseType>())).Returns(queryBuilder);
             queryManagerFactory.Setup(x => x.GetQueryExecutor(It.IsAny<DatabaseType>())).Returns(queryExecutor.Object);
 
-            SqlMetadataProvider<SqlConnection, SqlDataAdapter, SqlCommand> provider = new MsSqlMetadataProvider(runtimeConfigProvider, queryManagerFactory.Object, sqlMetadataLogger, runtimeConfigProvider.GetConfig().GetDefaultDataSourceName());
-            string tableprefix = provider.GetTablePrefix(databaseName, schemaName);
-            Assert.AreEqual(tableprefix, expectedPrefix);
+            SqlMetadataProvider<SqlConnection, SqlDataAdapter, SqlCommand> provider = new MsSqlMetadataProvider(
+                runtimeConfigProvider,
+                queryManagerFactory.Object,
+                sqlMetadataLogger,
+                runtimeConfigProvider.GetConfig().GetDefaultDataSourceName())
+            {
+                DatabaseName = databaseName
```

Comment:
This is not really testing the new code that you added in the constructor - to deduce the database name using ConnectionStringBuilder. Can you set this databaseName in the connection string property of the `runtimeConfigProvider` in some other ways like string replace on `InitialCatalog` for e.g.?

Reply:
A good test would be an integration tests where you test the scenario described by #1908 and ensure that error/messaging does not occur and results from both tables are returned as expected.

Reply:
> A good test would be an integration tests where you test the scenario described by #1908 and ensure that error/messaging does not occur and results from both tables are returned as expected.

i added an entity and table with collisions that match the scenario in #1908, they are used in a new case in the Find Api Tests, `FindOnTableWithNamingCollision`, this is just fixing the unit test that we already had to make sure it matches the new format.

Reply:
Why are you still setting the DatabaseName here 
`{
                DatabaseName = databaseName`
}` ? Cant you let the constructor do the job given that you are replacing the connectionstring accurately on line 119 above?

Reply:
As we discussed on teams, it turns out that we will never have more than one database per sqlmetadataprovider so we don't need to include database name in this way.